compiler/src/test/java/com/squareup/anvil/compiler/dagger/ProvidesMethodFactoryGeneratorTest.kt

```diff
@@ -2440,6 +2440,358 @@ public final class DaggerComponentInterface implements ComponentInterface {
     }
   }
 
+  @Test fun `a factory class is generated for provided properties`() {
+    /*
+package com.squareup.test;
+
+import dagger.internal.Factory;
+import dagger.internal.Preconditions;
+import javax.annotation.processing.Generated;
+
+@Generated(
+    value = "dagger.internal.codegen.ComponentProcessor",
+    comments = "https://dagger.dev"
+)
+@SuppressWarnings({
+    "unchecked",
+    "rawtypes"
+})
+public final class DaggerModule1_GetStringFactory implements Factory<String> {
+  private final DaggerModule1 module;
+
+  public DaggerModule1_GetStringFactory(DaggerModule1 module) {
+    this.module = module;
+  }
+
+  @Override
+  public String get() {
+    return getString(module);
+  }
+
+  public static DaggerModule1_GetStringFactory create(DaggerModule1 module) {
+    return new DaggerModule1_GetStringFactory(module);
+  }
+
+  public static String getString(DaggerModule1 instance) {
+    return Preconditions.checkNotNull(instance.getString(), "Cannot return null from a non-@Nullable @Provides method");
+  }
+}
+     */
+    compile(
+        """
+        package com.squareup.test
+        
+        import dagger.Module
+        import dagger.Provides
+        
+        @Module
+        class DaggerModule1 {
+          @get:Provides val string: String = "abc"
```

Comment:
Is the `get:` qualifier necessary? If so, what is the rationale for requiring it? I can't see an unqualified annotation having any other interpretation, so it seems like unnecessary boilerplate.

Reply:
Yes, Dagger requires that and I want to stay compatible to Dagger as close as possible.

Reply:
Not sure I understand. Wouldn't this code not work anyway if you removed Anvil?

Reply:
Correct, with plain Dagger `@Provides` would NOT work and you need to use `@get:Provides`. Using Anvil for this should be a 1 to 1 replacement and I don't want them to diverge.

Reply:
Sorry, I meant – would code that is using Anvil continue to work if anvil were removed? If not, then I don't understand the reasoning for keeping this limitation that presumably only exists because Dagger isn't aware of Kotlin.

Reply:
Yes, if you only use this feature and you enable the Dagger annotation processor, then the code would still work. https://github.com/square/anvil#experimental-dagger-factory-generation